liblwgeom/lwgeom_topo.c

```diff
@@ -616,46 +615,27 @@ _lwt_CheckEdgeCrossing( LWT_TOPOLOGY* topo,
     lwerror("Could not convert edge geometry to GEOS: %s", lwgeom_geos_errmsg);
     return -1;
   }
-  prepared_edge = GEOSPrepare( edgegg );
-  if ( ! prepared_edge ) {
-    lwerror("Could not prepare edge geometry: %s", lwgeom_geos_errmsg);
-    return -1;
-  }
   edgebox = lwgeom_get_bbox( lwline_as_lwgeom(geom) );
 
   /* loop over each node within the edge's gbox */
   nodes = lwt_be_getNodeWithinBox2D( topo, edgebox, &num_nodes,
                                             LWT_COL_NODE_ALL, 0 );
   LWDEBUGF(1, "lwt_be_getNodeWithinBox2D returned %d nodes", num_nodes);
   if ( num_nodes == -1 ) {
-    GEOSPreparedGeom_destroy(prepared_edge);
-    GEOSGeom_destroy(edgegg);
     lwerror("Backend error: %s", lwt_be_lastErrorMessage(topo->be_iface));
     return -1;
   }
   for ( i=0; i<num_nodes; ++i )
   {
     LWT_ISO_NODE* node = &(nodes[i]);
-    GEOSGeometry *nodegg;
-    int contains;
     if ( node->node_id == start_node ) continue;
     if ( node->node_id == end_node ) continue;
     /* check if the edge contains this node (not on boundary) */
-    nodegg = LWGEOM2GEOS( lwpoint_as_lwgeom(node->geom) , 0);
     /* ST_RelateMatch(rec.relate, 'T********') */
-    contains = GEOSPreparedContains( prepared_edge, nodegg );
-    GEOSGeom_destroy(nodegg);
-    if (contains == 2)
-    {
-      GEOSPreparedGeom_destroy(prepared_edge);
-      GEOSGeom_destroy(edgegg);
-      _lwt_release_nodes(nodes, num_nodes);
-      lwerror("GEOS exception on PreparedContains: %s", lwgeom_geos_errmsg);
-      return -1;
-    }
+    const POINT2D *pt = getPoint2d_cp(node->geom->point, 0);
```

Comment:
For the sake of consistency, could you move variable declarations to the top of their block? Same for https://github.com/bjornharrtell/postgis/blob/950bf4743c97b9f61b501914d4d36febb1de35cc/liblwgeom/lwgeom_topo.c#L3322 and https://github.com/bjornharrtell/postgis/blob/950bf4743c97b9f61b501914d4d36febb1de35cc/liblwgeom/lwgeom_topo.c#L5046

Reply:
hmm but that means I have to drop const. :( oh well.

Reply:
um.. no it doesn't :) C is wierd..